daffodil-udf/README.md

```diff
@@ -39,39 +39,82 @@ The class can provide as many User Defined Function classes as it wishes, as lon
 
 #### User Defined Function Classes
 
-The User Defined Function class must extend Daffodil's `UserDefinedFunction` class. This class will contain the actual functionality implementers wish to add to DPath Expressions. It must be annotated with the `UserDefinedFunctionIdentification` annotation class, with the name and namespaceURI fields filled in with how the UDF will be called from the schema. It must also implement a function named *evaluate* that Daffodil will call to execute the desired UDF functionality. *There is no support for overloaded or void evaluate functions*.
+The User Defined Function class must extend Daffodil's `UserDefinedFunction` class. This class will contain the actual functionality implementers wish to add to DFDL Expressions. It must be annotated with the `UserDefinedFunctionIdentification` annotation class, with the name and namespaceURI fields filled in with how the UDF will be called from the schema. It must also implement a function named *evaluate* that Daffodil will call to execute the desired UDF functionality. *There is no support for overloaded or void evaluate functions*.
+
+#### User Defined Function Exceptions
+
+Daffodil proves two exception classes for users that wish to throw an error from within their UDFs.
+* `UserDefinedFunctionProcessingError` can be thrown when the implementer wishes to cause backtracking during parsing
+* `UserDefinedFunctionFatalException` can be thrown when the implementer wishes processing to be aborted all together
+
+All other exceptions are treated as `UserDefinedFunctionFatalException`.
+
+### UDF Registration
+Once implemented, a UDF can be expected to have one of the following structure
+```
+// sample scala UDF with sbt structure
+// based on sample UDF in daffodil-udf/src/test/scala
+src/
+  main/
+    scala/
+      org/
+        sgoodudfs/
+          example/
+              StringFunctions/
+                StringFunctionsProvider.scala //UDF Provider with UDF classes declared within
+    resources/
+      META-INF/
+        services/
+          org.apache.daffodil.udf.UserDefinedFunctionProvider
+```
+or
+```
+// sample java UDF with generic structure
+// based on sample UDF in daffodil-udf/src/test/java
+src/
+  org/
+    jgoodudfs/
+      example/
+          StringFunctions/
+            StringFunctionsProvider.java //UDF Provider
+            Compare.java //UDF class
+            Replace.java //UDF class
+  META-INF/
+    services/
+      org.apache.daffodil.udf.UserDefinedFunctionProvider
+```
+Each UDF is registered by including the fully qualified name of its provider in the `META-INF/services/org.apache.daffodil.udf.UserDefinedFunctionProvider`. The META-INF folder must be accessible from the root of whatever paths are on the classpath, otherwise it won't be picked up by ServiceLoader.
```

Comment:
It's not clear from the way this is worded that org.apache...UserDefinedFunctionProvider is the name of a file that is supposed to be textual and contain the name. Say "in a text file named 'org.apache...UserDefinedFunctionProvider'"

Also I learned an interesting style idea when writing markdown which is https://rhodesmill.org/brandon/2012/one-sentence-per-line/. 

I'm not suggesting further change here in your file, but it's an idea I'm planning to embrace when writing markdown from here on.